### ISSUE TITLE
V3: Error Processing Gen3 Hybrid Inverter Data

### ISSUE THREAD

**Issue by joealden** [britkat1980/giv_tcp#240]
Data from my AIO + Gateway is flowing perfectly in V3, but data from my Gen3 Hybrid Inverter (for solar) isn't.

Here's the error shown in the add-on logs:

```
Inv3 - read - [ERROR] - Error processing Inverter data: ((<class 'TypeError'>, TypeError('list indices must be integers or slices, not str'), <traceback object at 0xffffac6bd5c0>), 871)
Inv3 - read - [ERROR] - inverter Update failed so using last known good data from cache: ((<class 'Exception'>, Exception('Process Data Failure'), <traceback object at 0xffffac6a3a40>), 1587)
Inv3 - read - [ERROR] - runAll2 Error processing registers: ('KeyError', 'read.py', 1766)
Inv3 - read - [ERROR] - 5 consecutive timeout errors in watch loop. Restarting modbus connection:
```

Any ideas? Let me know if you need any more info!

**Comment by britkat1980** [comment]
If you have no batteries associated with the Gen3 then this might be the cause and is fixed in 3.0.1. please update and report back.

**Comment by joealden** [comment]
Yep 3.0.1 has fixed it - thanks!
